src/components/nodes/subProcess/index.js

```diff
@@ -34,14 +39,12 @@ export default {
     setNodeProp(node, 'id', value.id);
     setNodeProp(node, 'name', value.name);
 
-    const oldConfig = JSON.parse(node.definition.config);
     const currentConfig = JSON.parse(value.config);
 
     setNodeProp(node, 'calledElement', currentConfig.calledElement);
 
     if (currentConfig.name !== value.name) {
-      if (currentConfig.name && (oldConfig.name === value.name || !oldConfig.name)) {
-        // SubProcessFormSelect automatically updated the name so set the new name here
+      if (subProcessFormSelected(value.name, node) && currentConfig.name) {
```

Comment:
I don't know if this helps with readability for me. I like what @johnblackspear suggested, taking this whole line and moving it into a function with a name like "nameChangedByFormSelect". The current name, for me, suggests that some form in a sub process was selected.